src/server/trpc.ts

```diff
@@ -0,0 +1,33 @@
+import {initTRPC} from '@trpc/server'
+import {TrpcContext} from 'app/api/trpc/[trpc]/route'
+import {transformer} from './transformer'
+
+const t = initTRPC.context<TrpcContext>().create({
+  transformer,
+  errorFormatter({shape}) {
+    return shape
+  },
+})
+
+/* 
+Interesting idea to auth a procedure instead of doing all of the manually checks for userToken in each router
+const isAuthed = t.middleware(async ({ ctx, next }) => {
+  if (!ctx.auth?.userId) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
+      message: "Not authenticated",
+    });
+  }
+
+  return next({
+    ctx: {
+      ...ctx,
+      auth: ctx.auth,
+    },
+  });
+});
+
+*/
```

Comment:
I saw an example of exporting an 'authed' Procedure so that way we wouldn't have to manually check for a token in every router that needs it. just add that logic in middleware here and we can know we are authed in the router where we use this procedure

```
const token = ctx?.userToken
 if (!token) return null
```